src/content/sidebar/index.tsx

```diff
@@ -1,10 +1,109 @@
 import * as ReactDOM from "react-dom"
+import retargetEvents from 'react-shadow-dom-retarget-events';
 import {SidebarWrapper} from "./sidebar-wrapper"
+import "@webcomponents/custom-elements"
 
-export const setupSidebar = () => {
-    const container = document.createElement('div');
-    container.classList.add('wd--inline-sidebar-container');
+class WikidataSidebar extends HTMLElement {
+  constructor() {
+    super();
+    let style = document.createElement('style');
+
+    // i'd rather source this from a seperate file, but I didn't see
+    // a way to do that inside the shadowDOM
+    style.textContent = `
```

Comment:
Presumably the whole wrapper component now is rendered in shadow-dom, so you can just import the css file inside that component?

also, I've set up emotion css-in-js, which should make it nicer to do the conditional css

Reply:
I managed to insert the css via `<link />`. I prefer to keep css and js separated.